Automattic/jetpack
# Allow supporting merge PRs with changelogger.

#### Changes proposed in this Pull Request:

In WooCommerce, we are hoping to use merge strategy along with squash for merging PRs. However, Jetpack changelogger currently does not support the merge strategy when param `add-pr-num` is provided. **Edit:** Now default merge message of format "Merge pull request #23123 from branch" is also supported.

This is because, in the git-log command, we list the latest commit for the changelog file, whereas what we want is the merge commit (which contains the PR number towards the in (#{pr_no}) format. This PR add supports to list merge commit, or a squash commit, depending upon what is available for the particular change, and therefore supports both strategies.

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

1. Create a new branch, say `temp`.
2. Create another branch with `temp` as base. Say `temp2`.
3. Add couple of commits to `temp2` branch.
4. Add a changelog.
5. Switch to `temp` branch, and merge the `temp2` branch with command like so: `git merge temp2 --no-ff`. For merge commit message, use something like `Allow supporting merge PRs with changelogger. (#27881)`. This is the default message format when the following setting is enabled:
<img width="936" alt="Screenshot 2022-12-12 at 3 19 47 PM" src="https://user-images.githubusercontent.com/7571618/207014553-f6747764-cef4-4a2e-a60f-53ef135f4b3c.png">
6. Run the changelogger command with params `--add-pr-num`. Check that the generated changelog has the PR number included.
7. Repeat the same, but this time in the merge commit message, use `Merge pull request #27881 from changelogger/merge` and verify that the change logger can pick the PR numbers.
